src/network/protocol/redis/command/network_protocol_redis_command_mget.c

```diff
@@ -0,0 +1,316 @@
+        goto end;
+    }
+
+    mget_command_context->keys[argument_index] = mget_command_context->key_data;
+    mget_command_context->key_data = NULL;
+end:
+    return true;
+}
+
+NETWORK_PROTOCOL_REDIS_COMMAND_FUNCPTR_COMMAND_END(mget) {
+    bool res;
+    bool return_res = false;
+    storage_db_entry_index_t *entry_index = NULL;
+
+    mget_command_context_t *mget_command_context = (mget_command_context_t*)protocol_context->command_context;
+
+    if (mget_command_context->has_error) {
+        char error_send_buffer[256], *error_send_buffer_start, *error_send_buffer_end;
+        size_t error_send_buffer_length;
+        error_send_buffer_length = sizeof(error_send_buffer);
+        error_send_buffer_start = error_send_buffer;
+        error_send_buffer_end = error_send_buffer_start + error_send_buffer_length;
+
+        error_send_buffer_start = protocol_redis_writer_write_simple_error(
+                error_send_buffer_start,
+                error_send_buffer_end - error_send_buffer_start,
+                mget_command_context->error_message,
+                (int)strlen(mget_command_context->error_message));
+
+        if (error_send_buffer_start == NULL) {
+            LOG_E(TAG, "buffer length incorrectly calculated, not enough space!");
+            goto end;
+        }
+
+        return_res = network_send(
+                channel,
+                error_send_buffer,
+                error_send_buffer_start - error_send_buffer) == NETWORK_OP_RESULT_OK;
+
+        goto end;
+    }
+
+    char send_buffer[64], *send_buffer_start, *send_buffer_end;
+    size_t send_buffer_length;
+
+    send_buffer_length = sizeof(send_buffer);
+    send_buffer_start = send_buffer;
+    send_buffer_end = send_buffer_start + send_buffer_length;
+
+    send_buffer_start = protocol_redis_writer_write_array(
+            send_buffer_start,
+            send_buffer_end - send_buffer_start,
+            mget_command_context->keys_count);
+
+    if (send_buffer_start == NULL) {
+        LOG_E(TAG, "buffer length incorrectly calculated, not enough space!");
+        goto end;
+    }
+
+    for(int key_index = 0; key_index < mget_command_context->keys_count; key_index++) {
+        storage_db_entry_index_status_t old_status;
+
+        entry_index = storage_db_get_entry_index(
+            db,
+            mget_command_context->keys[key_index]->key,
+            mget_command_context->keys[key_index]->key_length);
+
+        if (likely(entry_index)) {
+                // Try to acquire a reader lock until it's successful or the entry index has been marked as deleted
+                storage_db_entry_index_status_increase_readers_counter(
+                        entry_index,
+                        &old_status);
+
+                // TODO: ??
+                if (unlikely(old_status.deleted)) {
+                    entry_index = NULL;
+                }
+
+                send_buffer_start = protocol_redis_writer_write_argument_blob_start(
```

Comment:
because of the very long explanation above, the previous
```
if (likely(entry_index)) {
```

Should be terminated before this line and another exactly the same should be added there so if the entry_index has been deleted meanwhile it was being locked the code will flow as if there was no key available at all.